From 89c814868918f47de9d9ef1438b2799c38af070c Mon Sep 17 00:00:00 2001
From: Jamal Clarke <jamal_clarke@hotmail.co.uk>
Date: Tue, 13 May 2025 23:26:29 +0100
Subject: [PATCH] Fft windowing function timing

- Not isolated as the FFT task is the highest priority one therefore
  won't be interrupted
- Timing is for the time it takes to apply the current windowing
  function to the complex data, concerned with the time it takes to go
  from low to high
- Must apply the initial gpio debug patch prior to this one
---
 src/rtos/fft_rtos.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/rtos/fft_rtos.c b/src/rtos/fft_rtos.c
index f2e93d8..c92a4a4 100644
--- a/src/rtos/fft_rtos.c
+++ b/src/rtos/fft_rtos.c
@@ -2,6 +2,7 @@
 #include "extern_i2s_dma_data.h"
 #include "mic_data_processing.h"
 #include "fft_processing.h"
+#include "stm32g4xx_gpio_debug.h"
 
 #include "stm32g4xx_hal.h"
 #include "stm32g4xx_nucleo.h"
@@ -85,7 +86,9 @@ void fft_task_processing(void* pvParameters)
 			data_buffer[i * 2] = data_buffer[i * 2] - dc_signal;
 		}
 
+		deassert_gpio_debug_pin();
 		arm_mult_f32(data_buffer, complex_window_func, fft_input, FFT_DATA_SIZE * 2);
+		assert_gpio_debug_pin();
 
 		arm_cfft_f32(&arm_cfft, fft_input, inverse_fft, bit_reverse);
 		// ignore DC component, any gather real frequencies and Nyquist
-- 
2.49.0

